library/src/main/java/me/drakeet/multitype/MultiTypeAdapter.java

```diff
@@ -317,40 +316,24 @@ public final void onViewDetachedFromWindow(@NonNull ViewHolder holder) {
 
     @NonNull
     private ItemViewBinder getRawBinderByViewHolder(@NonNull ViewHolder holder) {
-        return typePool.getItemViewBinders().get(holder.getItemViewType());
+        return typePool.getItemViewBinder(holder.getItemViewType());
     }
 
 
     int indexInTypesOf(@NonNull Object item) throws BinderNotFoundException {
         int index = typePool.firstIndexOf(item.getClass());
         if (index != -1) {
             @SuppressWarnings("unchecked")
-            Linker<Object> linker = (Linker<Object>) typePool.getLinkers().get(index);
+            Linker<Object> linker = (Linker<Object>) typePool.getLinker(index);
             return index + linker.index(item);
         }
         throw new BinderNotFoundException(item.getClass());
     }
 
-
     private void checkAndRemoveAllTypesIfNeed(@NonNull Class<?> clazz) {
-        if (!typePool.getClasses().contains(clazz)) {
-            return;
-        }
-        Log.w(TAG, "You have registered the " + clazz.getSimpleName() + " type. " +
```

Comment:
Providing a boolean result at TypePool#unregister(class) is also a solution. Add a contains method is also OK, but is it necessary?